internal/cmd/templates/templates.go

```diff
@@ -52,21 +57,59 @@ func NewTemplateManager() (*TemplateManager, error) {
 		minimalTemplate:  minimalTmpl,
 		protocolTemplate: protocolTmpl,
 		browserTemplate:  browserTmpl,
+		fs:               fs,
 	}, nil
 }
 
 // GetTemplate selects the appropriate template based on the type
 func (tm *TemplateManager) GetTemplate(templateType string) (*template.Template, error) {
+	// First check built-in templates
 	switch templateType {
 	case MinimalTemplate:
 		return tm.minimalTemplate, nil
 	case ProtocolTemplate:
 		return tm.protocolTemplate, nil
 	case BrowserTemplate:
 		return tm.browserTemplate, nil
-	default:
-		return nil, fmt.Errorf("invalid template type: %s", templateType)
 	}
+
+	// Then check if it's a file path
+	if isFilePath(templateType) {
+		content, err := fsext.ReadFile(tm.fs, templateType)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read template file %s: %w", templateType, err)
+		}
+
+		tmpl, err := template.New(filepath.Base(templateType)).Parse(string(content))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse template file %s: %w", templateType, err)
+		}
+		return tmpl, nil
+	}
+
+	return nil, fmt.Errorf("invalid template type: %s", templateType)
+}
+
+// isFilePath checks if the given string looks like a file path
+// It handles both POSIX-style paths (./, ../, /) and Windows-style paths (C:\, \\, .\)
+func isFilePath(path string) bool {
+	// Check POSIX-style paths
+	if strings.HasPrefix(path, "./") ||
+		strings.HasPrefix(path, "../") ||
+		strings.HasPrefix(path, "/") {
+		return true
+	}
+
+	// Check Windows-style paths
+	if strings.HasPrefix(path, ".\\") ||
+		strings.HasPrefix(path, "..\\") ||
+		strings.HasPrefix(path, "\\") ||
+		strings.HasPrefix(path, "\\\\") || // UNC paths
+		(len(path) >= 2 && path[1] == ':') { // Drive letter paths like C:
+		return true
+	}
+
+	return false
 }
```

Comment:
This heuristic check can fail when the file's name is the same as a valid template type (e.g., browser). Another case it might fail when the file is provided with a relative path (e.g., --template=myFile). A better way might be to separate the template type and the file name. We can then use `fs.Exists` instead of a heuristic check for reliability.

Reply:
The implementation expects the file path to be either absolute or relative, so it always starts with `/`, `./` or `../`. I believe it's safe to assume that none of the built-in templates will have these prefixes. To use a template stored in `browser` file, user would need to specify it either as `./browser` or `/foo/bar/browser`.

Reply:
Yep, I see that users need to provide a file path like `./browser` for files and `browser` for a built-in template. However, distinguishing between a file and a built-in template just by checking a prefix like `./` can be easy to overlook and error-prone for users. It feels like we're mixing two concepts: template type and template source.

Reply:
I've added an explicit error message, that checks if a file with this name exists in current working directory and suggests the correct value.

Reply:
Thanks! Better 👍 I liked the way it says Google-like _"invalid template type "foo", did you mean ./foo?"_ 😄 But it still feels a little bit odd because the template type is a file path and users might have typed `browse` (without the `r` suffix), which would be confusing to tell them to type `./browser` :(

Reply:
I think it's would be too user-friendly to implement fuzzy matching with all files in the current directory 😃